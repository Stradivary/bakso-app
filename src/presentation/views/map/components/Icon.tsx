import L from "leaflet";
import markerPerson from "@/shared/assets/cust.svg";
import markerBakso from "@/shared/assets/seller.svg";

const iconPerson = new L.Icon({
  iconUrl: markerPerson,
  iconSize: new L.Point(32, 49),
});

const iconBakso = new L.Icon({
  iconUrl: markerBakso,
  iconSize: new L.Point(32, 49),
});

export { iconPerson, iconBakso };
