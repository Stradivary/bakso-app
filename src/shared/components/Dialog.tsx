import { Box, Drawer, Modal } from "@mantine/core";
import { useMediaQuery } from "@mantine/hooks";
import { PropsWithChildren } from "react";

export function Dialog({ opened, close, children }: PropsWithChildren<{ opened: boolean; close: () => void; }>) {

  const isMobile = useMediaQuery("(max-width: 768px)");

  if (!isMobile) {
    return <Modal opened={opened} onClose={close} size="sm" withCloseButton={false}>
      {children}
    </Modal>;
  }

  return <Drawer
    position="bottom"
    opened={opened}
    onClose={close}
    withCloseButton={false}
    withinPortal
    offset={8}
    styles={{
      content: {
        borderTopLeftRadius: 16,
        borderTopRightRadius: 16,
      },
    }}
  >
    <Box
      style={{
        width: 42,
        height: 4,
        background: "#DFE2E7",
        borderRadius: 8,
        flex: "none",
        order: 1,
        flexGrow: 0,
        margin: "16px auto 24px auto",
      }}>
    </Box>
    {children}

  </Drawer>;
}
