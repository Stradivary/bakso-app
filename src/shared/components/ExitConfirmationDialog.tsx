import { Stack, Center, Button, Text, Image } from "@mantine/core";

export function ExitConfirmationDialog({ role, handleExit, exitModalClose }: { role: string, handleExit: () => void; exitModalClose: () => void; }) {
  return <Stack>
    <Center>
      <Image src="/confirmation.png" alt="confirmation" w={100} h={100} />
    </Center>
    <Text ta="center">
      Dengan menutup halaman ini,
      {
        role === "seller"
          ? "Customer tidak akan bisa melihat lokasi Anda"
          : "anda akan keluar dari pantauan Tukang Bakso"
      }
    </Text>
    <Button onClick={handleExit} radius="lg" size="md">
      OK
    </Button>
    <Button onClick={exitModalClose} radius="lg" variant="outline">
      Batal
    </Button>
  </Stack>;
}
