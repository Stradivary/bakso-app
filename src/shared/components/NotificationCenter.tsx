import {
  ActionIcon,
  Button,
  Card,
  Indicator,
  Popover,
  ScrollArea,
  Stack,
  Text,
  Group,
} from "@mantine/core";
import { useDisclosure } from "@mantine/hooks";
import dayjs from "dayjs";
import { Bell, CheckCheck } from "lucide-react";
import { useCallback } from "react";
import { Notification } from "../hooks/useNotification";

export function NotificationCenter({
  notifications,
}: Readonly<{ notifications?: Notification[] }>) {
  const [opened, { toggle }] = useDisclosure();

  const handleNotificationClick = useCallback(
    async (notificationId: string) => {
      const notification = notifications?.find(
        (note) => note.id === notificationId,
      );
      if (!notification) return;
      notification.is_read = true;
    },
    [notifications],
  );

  const handleClearNotifications = useCallback(() => {
    notifications?.forEach((note) => (note.is_read = true));
  }, [notifications]);

  return (
    <Popover opened={opened} position="bottom-end" shadow="md">
      <Popover.Target>
        <Indicator
          disabled={notifications?.filter((note) => !note.is_read).length === 0}
          processing
          size={12}
        >
          <Button
            data-testid="notification-center"
            variant="white"
            size="compact-lg"
            onClick={() => toggle()}
          >
            <Bell size={24} />
          </Button>
        </Indicator>
      </Popover.Target>

      <Popover.Dropdown w={250} style={{ zIndex: 800 }}>
        <Stack gap="xs">
          <Group justify="space-between" align="center">
            <Text fw={500}>Notifikasi</Text>
            <ActionIcon
              variant="transparent"
              size="xs"
              onClick={() => handleClearNotifications()}
            >
              <CheckCheck size={20} />
            </ActionIcon>
          </Group>
          <ScrollArea style={{ height: 300 }}>
            <Stack gap="xs">
              {notifications?.length === 0 ? (
                <Text c="dimmed" ta="center">
                  Tidak ada notifikasi
                </Text>
              ) : (
                notifications?.map((notification) => {
                  return (
                    <Card
                      key={notification.id}
                      p="xs"
                      withBorder
                      onClick={() => handleNotificationClick(notification.id)}
                      style={{
                        backgroundColor: notification.is_read
                          ? undefined
                          : "#f0f9ff",
                        cursor: "pointer",
                      }}
                    >
                      <Text size="sm">
                        Seorang pelanggan mencolekmu:{" "}
                        {notification.buyer_name ?? "Unknown User"}
                      </Text>
                      <Text size="xs" c="dimmed">
                        {notification.expiry_at
                          ? dayjs(notification.expiry_at).format(
                              "DD MMMM YYYY\n HH:mm",
                            )
                          : "No date"}
                      </Text>
                    </Card>
                  );
                })
              )}
            </Stack>
          </ScrollArea>
        </Stack>
      </Popover.Dropdown>
    </Popover>
  );
}
