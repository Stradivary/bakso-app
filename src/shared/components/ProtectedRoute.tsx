import { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { useAuth } from "../hooks/useAuth";

export const ProtectedRoute: React.FC<{ children: JSX.Element; }> = ({
  children,
}) => {
  const { isAuthenticated } = useAuth();
  const navigate = useNavigate();
  useEffect(() => {
    if (!isAuthenticated) {

      navigate('/login')
    }
  }, [isAuthenticated, navigate])
  return children;
};
