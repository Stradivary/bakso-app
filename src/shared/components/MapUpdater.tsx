import { Stack, Text } from "@mantine/core";
import { LatLng, Point } from "leaflet";
import { useEffect } from "react";
import { Marker, Tooltip } from "react-leaflet";
import { useLocation } from "../hooks/useLocation";
import { User } from "../models/BroadcastMapModel";
import { iconBakso, iconPerson } from "./Icon";

const UPDATE_INTERVAL = 2 * 1000;

export const MapUpdater = ({
  disabled,
  userId,
  userRole,
  updateLocation,
}: {
  disabled: boolean,
  userId: string,
  userRole: string,
  updateLocation: (userId: string, location: LatLng) => void;
}
) => {
  const { location } = useLocation();

  useEffect(() => {
    if (userRole === 'seller' && !disabled) {
      const updateInterval = setInterval(() => {
        updateLocation(userId, new LatLng(location?.latitude ?? 0, location?.longitude ?? 0));
      }, UPDATE_INTERVAL);

      return () => clearInterval(updateInterval);
    }
  }, [userRole, updateLocation, location?.latitude, location?.longitude, disabled]);

  return null;
};


export function TargetMark({ handleClick, nearbyUser }: {
  nearbyUser: User,
  handleClick: () => L.LeafletMouseEventHandlerFn | undefined,
}) {
  return <Marker
    position={new LatLng(nearbyUser?.location?.lat ?? 0, nearbyUser?.location?.lng ?? 0)}
    icon={nearbyUser.role === 'seller' ? iconBakso : iconPerson}
    eventHandlers={{
      click: handleClick,
    }}
  >
    <Tooltip>
      <Stack gap={4}>
        <Text>{nearbyUser.userName}</Text>
        <Text c="gray">{nearbyUser.role === 'seller' ? 'Pedagang Bakso' : 'Pelanggan'}</Text>
      </Stack>
    </Tooltip>
  </Marker>;
}


export function UserMarker({ location, userRole }: { location: { latitude: number; longitude: number; }; userRole: string; }) {
  return (<Marker
    position={new LatLng(location?.latitude ?? 0, location?.longitude ?? 0)}
    icon={userRole === 'seller' ? iconBakso : iconPerson}>
    <Tooltip permanent direction='bottom' offset={new Point(0, 12)}>
      Lokasi Anda
    </Tooltip>
  </Marker>);
}