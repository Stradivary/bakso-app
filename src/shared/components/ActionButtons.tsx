import { Badge, Box, Button, Flex, Group, Stack } from "@mantine/core";
import { LocateFixed, X } from "lucide-react";
import { NotificationCenter } from "./NotificationCenter";
import { ActionButtonProps } from "../models/actionButtons.types";

export const ActionButtons = ({
  onExit,
  onRecenter,
  role,
  name,
  notifications,
}: ActionButtonProps) => (
  <Box
    style={{
      zIndex: 50,
      position: "absolute",
      top: 0,
      right: 0,
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-end",
      padding: 8,
    }}
  >
    <Flex direction="row">
      <Group justify="end" align="start" m={8} gap={6} style={{
        userSelect: 'none',
        pointerEvents: 'painted'
      }}>
        <Badge color={role === "seller" ? "green" : "blue"}>
          {role === "seller" ? "Tukang Bakso" : "Customer"}
        </Badge>
        <Badge variant="filled" color="teal">
          {name}
        </Badge>
      </Group>
      <Stack m={8} style={{ flexGrow: 1 }} align="end">
        <Button onClick={onRecenter} variant="white" size="compact-lg" w={50}>
          <LocateFixed size={20} />
        </Button>
        <Button
          color="red"
          onClick={onExit}
          variant="white"
          size="compact-lg"
          w={50}
        >
          <X size={20} />
        </Button>
        {role === "seller" && (
          <NotificationCenter notifications={notifications} />
        )}
      </Stack>

    </Flex>
  </Box>
);
