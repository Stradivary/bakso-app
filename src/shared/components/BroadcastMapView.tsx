import { Badge, Button, Center, Image, Stack, Text } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import { modals } from '@mantine/modals';
import L, { LatLng, Point } from 'leaflet';
import React, { useEffect } from 'react';
import { MapContainer, Marker, Popup, TileLayer, Tooltip } from 'react-leaflet';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../hooks/useAuth';
import { useLocation } from '../hooks/useLocation';
import { calculateDistance, useLocationTracking } from '../hooks/useTracker';
import { ActionButtons } from './ActionButtons';
import { Dialog } from './Dialog';
import { iconBakso, iconPerson } from "./Icon";

const UPDATE_INTERVAL = 5 * 1000; // Update interval in milliseconds (5 seconds)

// Custom hook to handle map updates
const MapUpdater = ({
  userRole,
  updateLocation,
}: {
  userRole: string,
  updateLocation: (location: L.LatLng) => void;
}
) => {
  const { location } = useLocation();

  useEffect(() => {
    if (userRole === 'seller') {
      const updateInterval = setInterval(() => {
        updateLocation(new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0));
      }, UPDATE_INTERVAL);

      return () => clearInterval(updateInterval);
    }
  }, [userRole, updateLocation, location?.latitude, location?.longitude]);

  return null;
};

const BroadcastMapView: React.FC = () => {
  const navigate = useNavigate();
  const { session, logout } = useAuth();

  const userId = session?.user?.id as string;
  const userRole = sessionStorage.getItem('role') as 'seller' | 'buyer';
  const [exitModalOpened, { close: exitModalClose, open: openModal }] = useDisclosure();

  const { location } = useLocation();


  const {
    nearbyUsers,
    updateLocation,
    pingSeller,
    deactivateUser,
    notifications
  } = useLocationTracking(userId, userRole, new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0), session?.user?.user_metadata?.name ?? '');


  const [mapRef, setMapRef] = React.useState<L.Map | null>(null);

  const handleRecenter = React.useCallback(() => {
    if (mapRef && location) {
      mapRef.setView(new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0), mapRef.getZoom(), {
        animate: true,
        duration: 1,
      });
    }
  }, [mapRef, location]);

  const handleExit = React.useCallback(() => {
    deactivateUser();
    logout();
    sessionStorage.removeItem('role');
    exitModalClose();
  }, [deactivateUser, exitModalClose, logout, navigate]);

  if (!location) {
    return (
      <div>
        <Text>Getting your location...</Text>
      </div>
    );
  }

  return (
    <>
      <MapContainer
        key={location.toString()}
        ref={(map) => setMapRef(map)}
        center={new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0) ?? new LatLng(0, 0)}
        zoom={16}
        minZoom={13}
        maxZoom={18}
        style={{ height: "100dvh", width: '100%', zIndex: 0 }}
      >
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />


        {userRole === 'seller' && (
          <MapUpdater userRole={userRole} updateLocation={updateLocation} />
        )}

        {/* Render user's current location */}
        <Marker position={new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0)} icon={userRole === 'seller' ? iconBakso : iconPerson}>
          <Tooltip permanent direction='bottom' offset={new Point(0, 12)}>
            Lokasi
            Anda
          </Tooltip>
        </Marker>

        {/* Render nearby users */}
        {nearbyUsers.map((nearbyUser) => {
          return (
            <Marker
              key={`${nearbyUser.user_id} ${nearbyUser.role}`}
              position={new L.LatLng(nearbyUser?.location?.lat ?? 0, nearbyUser?.location?.lng ?? 0)}
              icon={nearbyUser.role === 'seller' ? iconBakso : iconPerson}
              eventHandlers={{
                click: () => {
                  if (nearbyUser.role === "seller") {
                    const buyerLocation = new L.LatLng(location?.latitude ?? 0, location?.longitude ?? 0);
                    const sellerLocation = new L.LatLng(
                      nearbyUser.location.lat,
                      nearbyUser.location.lng
                    );
                    const distanceInMeters = calculateDistance(buyerLocation, sellerLocation);
                    const walkingSpeed = 1.4; // meters per second
                    const estimatedTimeInMinutes = Math.round((distanceInMeters / walkingSpeed) / 60);

                    modals.openConfirmModal({
                      title: "Pesan Bakso",
                      children: (
                        <>
                          <Text size="sm">
                            üçú Apakah Anda ingin memesan bakso dari {nearbyUser.userName}?
                          </Text>
                          <Text size="xs" c="dimmed">
                            Estimasi waktu: {estimatedTimeInMinutes} menit berjalan kaki
                          </Text>
                        </>
                      ),
                      labels: { cancel: "Batal", confirm: "Pesan" },
                      onConfirm: () => {
                        pingSeller?.(nearbyUser.user_id);
                        modals.open({
                          title: "Pesan Bakso",
                          children: "Pesan bakso telah dikirim",
                          color: "blue",
                        });
                      },
                    });
                  }
                },
              }}
            >
              <Popup>
                <div   >
                  <Text>{nearbyUser.userName}</Text>
                  <Text c="gray">{nearbyUser.role === 'seller' ? 'Pedagang Bakso' : 'Pelanggan'}</Text>

                </div>
              </Popup>
            </Marker>
          );
        })}
      </MapContainer>

      <ActionButtons
        role={userRole}
        isLoading={false}
        onExit={() => {
          openModal();
        }}
        notifications={notifications}
        onRecenter={handleRecenter}
      >
        <Badge variant='filled' color='teal'>
          {session?.user?.user_metadata?.name}
        </Badge>
      </ActionButtons>

      <Dialog opened={exitModalOpened} close={exitModalClose}>
        <Stack>
          <Center>
            <Image src="/confirmation.png" alt="confirmation" w={100} h={100} />
          </Center>
          <Text ta="center">
            Dengan menutup halaman ini, kamu akan keluar dari pantauan Tukang
            Bakso
          </Text>
          <Button onClick={handleExit} radius="lg" size="md">
            OK
          </Button>
          <Button onClick={exitModalClose} radius="lg" variant="outline">
            Batal
          </Button>
        </Stack>
      </Dialog>
    </>
  );
};

export default BroadcastMapView;