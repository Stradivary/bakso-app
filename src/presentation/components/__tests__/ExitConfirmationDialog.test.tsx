import { MantineProvider } from "@mantine/core";
import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { ExitConfirmationDialog } from "../../views/map/components/ExitConfirmationDialog";

describe("ExitConfirmationDialog", () => {
  const handleExit = vi.fn();
  const exitModalClose = vi.fn();

  it("renders seller message correctly", () => {
    const { container } = render(
      <MantineProvider>
        <ExitConfirmationDialog
          role="seller"
          handleExit={handleExit}
          exitModalClose={exitModalClose}
        />
      </MantineProvider>,
    );
    expect(
      screen.getByText(/Customer tidak akan bisa melihat lokasi Anda/),
    ).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("renders buyer message correctly", () => {
    const { container } = render(
      <MantineProvider>
        <ExitConfirmationDialog
          role="buyer"
          handleExit={handleExit}
          exitModalClose={exitModalClose}
        />
      </MantineProvider>,
    );
    expect(
      screen.getByText(/anda akan keluar dari pantauan Tukang Bakso/),
    ).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("calls handleExit when OK button is clicked", () => {
    const { container } = render(
      <MantineProvider>
        <ExitConfirmationDialog
          role="seller"
          handleExit={handleExit}
          exitModalClose={exitModalClose}
        />
      </MantineProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
