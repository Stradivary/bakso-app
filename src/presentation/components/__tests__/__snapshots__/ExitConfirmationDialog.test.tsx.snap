// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExitConfirmationDialog > calls handleExit when OK button is clicked 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
  >
    <div
      class="m_4451eb3a mantine-Center-root"
    >
      <img
        alt="confirmation"
        class="m_9e117634 mantine-Image-root"
        src="/confirmation.png"
      />
    </div>
    <p
      class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      style="text-align: center;"
    >
      Dengan menutup halaman ini,
      Customer tidak akan bisa melihat lokasi Anda
    </p>
    <button
      class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-size="md"
      style="--button-height: var(--button-height-md); --button-padding-x: var(--button-padding-x-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-lg); --button-color: var(--mantine-color-white);"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        >
          OK
        </span>
      </span>
    </button>
    <button
      class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-variant="outline"
      style="--button-radius: var(--mantine-radius-lg); --button-bg: transparent; --button-hover: var(--mantine-color-blue-outline-hover); --button-color: var(--mantine-color-blue-outline); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        >
          Batal
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ExitConfirmationDialog > renders buyer message correctly 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
  >
    <div
      class="m_4451eb3a mantine-Center-root"
    >
      <img
        alt="confirmation"
        class="m_9e117634 mantine-Image-root"
        src="/confirmation.png"
      />
    </div>
    <p
      class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      style="text-align: center;"
    >
      Dengan menutup halaman ini,
      anda akan keluar dari pantauan Tukang Bakso
    </p>
    <button
      class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-size="md"
      style="--button-height: var(--button-height-md); --button-padding-x: var(--button-padding-x-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-lg); --button-color: var(--mantine-color-white);"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        >
          OK
        </span>
      </span>
    </button>
    <button
      class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-variant="outline"
      style="--button-radius: var(--mantine-radius-lg); --button-bg: transparent; --button-hover: var(--mantine-color-blue-outline-hover); --button-color: var(--mantine-color-blue-outline); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        >
          Batal
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ExitConfirmationDialog > renders seller message correctly 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
  >
    <div
      class="m_4451eb3a mantine-Center-root"
    >
      <img
        alt="confirmation"
        class="m_9e117634 mantine-Image-root"
        src="/confirmation.png"
      />
    </div>
    <p
      class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      style="text-align: center;"
    >
      Dengan menutup halaman ini,
      Customer tidak akan bisa melihat lokasi Anda
    </p>
    <button
      class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-size="md"
      style="--button-height: var(--button-height-md); --button-padding-x: var(--button-padding-x-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-lg); --button-color: var(--mantine-color-white);"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        >
          OK
        </span>
      </span>
    </button>
    <button
      class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-variant="outline"
      style="--button-radius: var(--mantine-radius-lg); --button-bg: transparent; --button-hover: var(--mantine-color-blue-outline-hover); --button-color: var(--mantine-color-blue-outline); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        >
          Batal
        </span>
      </span>
    </button>
  </div>
</div>
`;
